lib/matplotlib/animation.py

```diff
@@ -933,7 +924,7 @@ def __init__(self, fps=30, codec=None, bitrate=None, extra_args=None,
 
         if self.default_mode not in ['loop', 'once', 'reflect']:
             self.default_mode = 'loop'
-            warnings.warn("unrecognized default_mode: using 'loop'")
+            _log.warning("unrecognized default_mode: using 'loop'")
```

Comment:
Since this comes from the caller, I'm not sure it should be logged instead of a warning.

Reply:
Attn: @dopplershift, what do you think: should any of the warnings in animation be converted to _log.warning?

Reply:
It's not clear to me what difference the change makes. I guess warnings from `warnings.warn` don't repeat? In that case I think using the log might be better in that regard.

Something else I'm missing?

Reply:
@dopplershift  `warnings.warns` can't be fine-tuned by the user to appear or not appear.  So, its pretty easy to suppress all matplotlib-specific logging.warning() messages.  I chose to use logging here just because it seemed like a non-critical warning some folks would want to suppress.  OTOH, I'm not sure why I picked on this example.  Happy to do either way..

Reply:
I think the default logging config does not print a source file or line number.

This warning comes from an invalid input from the user. Actually, I'm not sure why it doesn't just raise an error.

Reply:
Just to re-iterate, I'm not trying to make decisions for the original authors of the modules.  All I did was rip out `verbose` and replace with `logging`.  If the actual contents of the messages need to be changed, or the level/mechanism is wrong, then we can do that as needed.

Reply:
I think I'm fine with the changes.

Reply:
I have no objection to the changes, either.
One additional point of information: warnings can be turned into errors, which can be very useful when debugging.